packages/paste-core/components/form/src/FormRadio.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {Text} from '@twilio-paste/text';
+import {ScreenReaderOnly} from '@twilio-paste/screen-reader-only';
+import {BackgroundColor} from '@twilio-paste/style-props';
+import {SiblingBox} from './shared/SiblingBox';
+
+export interface FormRadioProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  id: string;
+  value: string;
+  name?: string;
+  defaultIsChecked?: boolean;
+  isChecked?: boolean;
+  isFullWidth?: boolean;
+  isDisabled?: boolean;
+  isInvalid?: boolean;
+  helpText?: string | React.ReactNode;
+}
+
+const FormRadio = React.forwardRef<HTMLInputElement, FormRadioProps>(
+  (
+    {
+      id,
+      name,
+      value,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      defaultIsChecked,
+      isChecked,
+      isFullWidth,
+      isDisabled,
+      isInvalid,
+      onChange,
+      onBlur,
+      onFocus,
+      children,
+      helpText,
+      ...props
+    },
+    ref
+  ) => {
+    let dotBackgroundColor = null as BackgroundColor;
+    if (isChecked) {
+      dotBackgroundColor = 'colorBackgroundBody';
+    }
+    if (isDisabled && isChecked) {
+      dotBackgroundColor = 'colorBackgroundDarker';
+    }
+
+    return (
+      <Box
+        {...safelySpreadBoxProps(props)}
+        as="label"
+        display="inline-flex"
+        verticalAlign="top"
+        htmlFor={id}
+        alignItems="flex-start"
+        width={isFullWidth ? '100%' : undefined}
+        cursor={isDisabled ? 'not-allowed' : 'pointer'}
+      >
+        <ScreenReaderOnly
```

Comment:
I meant wrap an input with it
```
<ScreenReaderOnly>
  <input {...props} />
</ScreenReaderOnly>
```

Reply:
Doing so wrap the `input` in a `span` which makes the adjacent pseudo styles like `input[type=${type}]:focus + &` not work.

Reply:
ooooooooooooooooooooooooooooooohhhhh, Ok we need different typings then

Reply:
Resolved by using a styled `input`